### PR TITLE
Vendor order view and Confirmation Modals

### PR DESCRIPTION
**Pull request by jondubin**
Merged both the order modal and order view into a single branch since they share templating code

The Vendor now has an order view that he can use to view and approve/deny orders. The Merchant also has confirmation modals for ordering.

### CODE REVIEW

File: app/merchant/views.py [hunk 1]
```diff
@@ -81,9 +83,23 @@ def order_items():
...
+    return render_template(
+        'merchant/manage_cart.html',
+        vendor_items_dict=vendor_items_dict,
+        confirm_order=confirm_order,
+        vendor=vendor,
+        get_total_price=CartItem.get_total_price
+    )
 
 
 @merchant.route('/items/<int:listing_id>')
```
Comment (abhisuri97) [inline app/merchant/views.py:105]: Can we go ahead and merge in the listing info method?

File: app/templates/merchant/manage_cart.html [hunk 2]
```diff
@@ -1,5 +1,48 @@
...
+            <th>Unit Price</th>
+            <th>Total</th>
+        </tr>
+        </thead>
+        <tbody>
+        {% for cart_item in cart_items %}
+            <tr>
+                <td>{{ cart_item.listing.name }}</td>
+                <td>{{ cart_item.quantity| int() }} {{ cart_item.listing.category.unit }}</td>
+                <td>${{ '%0.2f' | format(cart_item.listing.price) }} / {{ cart_item.listing.category.unit }}</td>
```
Comment (abhisuri97) [inline app/templates/merchant/manage_cart.html]: im seeing `'%0.2f' | format(...)` a lot. is there a way to just generalize it under a more concise method? (not really something to worry about, just something to consider)

File: app/templates/merchant/orders.html [hunk 3]
```diff
@@ -0,0 +1,142 @@
+{% extends 'layouts/base.html' %}
```
Comment (abhisuri97) [inline app/templates/merchant/orders.html:1]: Add a filter for Pending, Active, Approved, and Rejected Orders.

File: app/templates/merchant/orders.html [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    {% set vendor_info = order.get_vendor_info() %}
+    {% if hide %}
+        {% set display = "display:none" %}
+    {% endif %}
+    <div class="ui segment clearing order" style="{{ display }}" id="order-{{ order.id }}">
+        <div class="ui left floated basic segment" style="margin-bottom: 0px">
+            <h3 class="ui header">
+                Order #{{ order.id }} - {{ vendor_info['company_name'] }} {{ status_label(order.status) }}
+                <div class="sub header">
+                    Placed on {{ order.get_date() }}
```
Comment (abhisuri97) [inline app/templates/merchant/orders.html]: Also put a time of day here

File: app/templates/merchant/orders.html [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+        </div>
+        <div class="ui right floated basic segment">
+            <div class="ui list right">
+                <div class="item"><b>Company name: </b> {{  vendor_info['company_name'] }}</div>
+                {% if vendor_info['full_name'] %}
+                    <div class="item"><b>Vendor name: </b> {{  vendor_info['full_name'] }}</div>
+                {% endif %}
+                {% if vendor_info['email'] %}
+                    <div class="item"><b>Email: </b> {{  vendor_info['email'] }}</div>
```
Comment (abhisuri97) [inline app/templates/merchant/orders.html]: add `<a href="mailto: {{ vendor_info['email'] }}">` for easy click to email

File: app/templates/merchant/orders.html [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+                Order #{{ order.id }} - {{ vendor_info['company_name'] }} {{ status_label(order.status) }}
+                <div class="sub header">
+                    Placed on {{ order.get_date() }}
+                </div>
+            </h3>
+            <div><b>Total:</b> {{ order.get_total() }}</div>
+            <div style="margin-top: 5px; margin-bottom: 0px"><a class="toggle-details">Show purchases</a></div>
+
+        </div>
+        <div class="ui right floated basic segment">
```
Comment (abhisuri97) [inline app/templates/merchant/orders.html:48]: Future implementation: Add button for easy Reorder

File: app/merchant/views.py (hunk 1 above)
Comment (jondubin) [inline app/merchant/views.py:105]: You mean Hunter's branch, `listing-view`?

File: app/templates/merchant/manage_cart.html (hunk 2 above)
Comment (jondubin) [inline app/templates/merchant/manage_cart.html]: Good call. I added a `format_price` function and injected it into Jinja's global environment.

File: app/templates/merchant/orders.html (hunk 6 above)
Comment (jondubin) [inline app/templates/merchant/orders.html:48]: Noted

File: app/templates/merchant/orders.html (hunk 5 above)
Comment (jondubin) [inline app/templates/merchant/orders.html]: Done, also fixed elsewhere

File: app/merchant/views.py (hunk 1 above)
Comment (abhisuri97) [inline app/merchant/views.py:105]: Yeah its been open for about a month or so. Note for merge later

File: app/templates/merchant/orders.html (hunk 4 above)
Comment (jondubin) [inline app/templates/merchant/orders.html]: Added

File: app/merchant/views.py (hunk 1 above)
Comment (jondubin) [inline app/merchant/views.py:105]: Yep, I worked on it yesterday to fix the admin page bug, and also to give admin's access to that view. After this PR is merged let's get that one in as well

File: app/templates/merchant/orders.html (hunk 3 above)
Comment (jondubin) [inline app/templates/merchant/orders.html:1]: Done

Comment (abhisuri97) [inline app/templates/merchant/orders.html:1]: confirmed working on my end

**Pull request merged**